### PR TITLE
Add filtering to PeopleList

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-13642
affects: Student
release note: Added filtering People by base enrollment type

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Added filtering People by base enrollment type

#### Affected Apps: Student

[MBL-13642](https://instructure.atlassian.net/browse/MBL-13642)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/People/PeopleListViewController.swift:191:76: Comma Spacing Violation: There should be no space before and one after any comma. (comma)
    Linting 'UITestHelpers.swift' (346/644)
    Linting 'FileUploadContext.swift' (347/644)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/People/PeopleListViewController.swift [hunk 1]
```diff
@@ -65,15 +69,27 @@ public class PeopleListViewController: UIViewController, ColoredNavViewProtocol
...
-        users.refresh()
         if context.contextType == .course {
             course.refresh()
         } else {
             group.refresh()
         }
+        users.refresh()
+
+        if context.contextType == .course {
+            env.api.makeRequest(GetConversationRecipientsRequest(search: "", context: context.canvasContextID, includeContexts: true)) { [weak self] recipients, _, _ in
```
Comment (armstrongnate) [inline Core/Core/People/PeopleListViewController.swift]: Is pagination going to be an issue here? This won't get all the recipients so the enrollment types array might be incomplete. You could use `env.api.exhaust` but that seems a bit heavy for large courses.

File: Core/Core/People/PeopleListViewController.swift [hunk 2]
```diff
@@ -171,11 +229,7 @@ class PeopleListCell: UITableViewCell {
         avatarView.url = user?.avatarURL
         nameLabel.text = user?.name
         let roles = user?.enrollments?.compactMap { $0.formattedRole }.sorted() ?? []
-        if #available(iOS 13, *) {
-            rolesLabel.text = ListFormatter.localizedString(from: roles)
-        } else {
-            rolesLabel.text = roles.joined(separator: ", ")
-        }
+        rolesLabel.text = ListFormatter.localizedString(from: roles)
```
Comment (armstrongnate) [inline Core/Core/People/PeopleListViewController.swift]: This doesn't require iOS 13 after all, huh?

File: Core/Core/Views/ListFormatter.swift [hunk 3]
```diff
@@ -27,6 +27,9 @@ public struct ListFormatter {
     ///
     /// See http://cldr.unicode.org/development/development-process/design-proposals/list-formatting
     public static func localizedString(from list: [String], conjunction: Conjunction = .and) -> String {
+        if #available(iOS 13, *), conjunction == .and {
```
Comment (armstrongnate) [inline Core/Core/Views/ListFormatter.swift]: Oh I see

File: Core/Core/People/PeopleListViewController.swift (hunk 1 above)
Comment (vanwagonet) [reply Core/Core/People/PeopleListViewController.swift]: this only returns the top level groups, no individuals, for a course, I've never seen more than 6 results.

File: Core/Core/People/PeopleListViewController.swift (hunk 2 above)
Comment (vanwagonet) [reply Core/Core/People/PeopleListViewController.swift]: We have our own `ListFormatter`, and I modified it to use the `Foundation` one when available.

File: Core/Core/People/PeopleListViewController.swift (hunk 1 above)
Comment (armstrongnate) [reply Core/Core/People/PeopleListViewController.swift]: Okay that makes sense.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018599&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018601&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018602&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**